sale_order_line_price_history/models/sale_order_line.py

```diff
@@ -9,3 +10,19 @@ class SaleOrderLine(models.Model):
     # In core this a related field. We need to trigger its value on view, so we can
     # have it even when we're in a NewId
     order_partner_id = fields.Many2one(depends=["product_id"])
+
+    def _compute_already_sold(self):
+        for line in self:
+            domain = [
+                ("product_id", "=", line.product_id.id),
+                ("state", "in", ["sale", "done"]),
+                ("order_partner_id", "child_of", line.order_id.partner_id.commercial_partner_id.ids),
+            ]
+            order_lines = line.env["sale.order.line"].search(domain, limit=1)
```

Comment:
I think this could be optimized through a read_group before the loop

Reply:
Done 👍